proto/penumbra/penumbra/core/governance/v1alpha1/governance.proto

```diff
@@ -212,7 +218,12 @@ message Proposal {
   // It optionally contains a reference to a commit which contains code to upgrade the chain.
   message Signaling {
     // The commit to be voted upon, if any is relevant.
-    optional string commit = 1;
+    Commit commit = 1;
+  }
+
+  // The sha1 hash of a git commit, used to indicate a specific version of the software.
+  message Commit {
+      string commit = 1;
```

Comment:
Alternatively, this could just be `string commit = 1` rather than a submessage, with the empty string meaning `None`.